app/main.py

```diff
@@ -1 +1,44 @@
-# write your code here
+class SoftwareEngineer:
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills.extend(["JavaScript", "HTML", "CSS"])
```

Comment:
You use `extend()` method to add multiple values to a single list, which is a good practice according to the checklist point 1. However, the `extend()` method should be used in the `learn_skill()` method of the `SoftwareEngineer` class, not in the `__init__()` method of each subclass. It would be more efficient and maintainable if you add the skills for each specific type of engineer in the `learn_skill()` method of each subclass, instead of doing it in the `__init__()` method.